extensions/ql-vscode/src/query-history.ts

```diff
@@ -611,7 +611,15 @@ export class QueryHistoryManager extends DisposableObject {
       await this.refreshTreeView();
     });
 
+    const variantAnalysisRemovedSubscription = this.variantAnalysisManager.onVariantAnalysisRemoved(async (variantAnalysis) => {
+      const item = this.treeDataProvider.allHistory.find(i => i.t === 'variant-analysis' && i.historyItemId === variantAnalysis.id.toString());
```

Comment:
```suggestion
      const item = this.treeDataProvider.allHistory.find(i => i.t === 'variant-analysis' && i.variantAnalysis.id === variantAnalysis.id);
```

The `historyItemId` is basically a randomly generated unique ID, so unfortunately you can't compare against it. To find the right item you want to check the variant analysis ID.